packages/ldc/ldc-manual-idgen.patch

```diff
@@ -0,0 +1,1095 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 336bbdbc..866b0f9e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -269,15 +269,15 @@ endmacro()
+ #
+ # Build idgen.
+ #
+-build_idgen(${DDMDFE_PATH}/idgen.d ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/idgen${CMAKE_EXECUTABLE_SUFFIX}  ${DDMD_DFLAGS} "" "")
++#build_idgen(${DDMDFE_PATH}/idgen.d ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/idgen${CMAKE_EXECUTABLE_SUFFIX}  ${DDMD_DFLAGS} "" "")
+ # Run idgen.
+ add_custom_command(
+     OUTPUT
+         ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/id.d
+         ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/id.h
+-    COMMAND ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/idgen  #provide full path to avoid clash with idgen on path
++    COMMAND /bin/cp  #provide fake command and copy the pre-generated files later
+     WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}
+-    DEPENDS ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/idgen${CMAKE_EXECUTABLE_SUFFIX}
++#    DEPENDS ${PROJECT_BINARY_DIR}/${DDMDFE_PATH}/idgen${CMAKE_EXECUTABLE_SUFFIX}
```

Comment:
This is a D file that generates `id.d` and `id.h` that is supposed to be built on the host and generates `id.d` and `id.h`.  Rather than add a dependency to yet another D compiler, I just ran `idgen` and manually add those two files to this patch below.